src/realm/alloc_slab.cpp

```diff
@@ -637,12 +637,14 @@ char* SlabAlloc::do_translate(ref_type) const noexcept
 
 int SlabAlloc::get_committed_file_format_version() const noexcept
 {
-    if (m_mappings.size()) {
-        // if we have mapped a file, m_mappings will have at least one mapping and
-        // the first will be to the start of the file. Don't come here, if we're
-        // just attaching a buffer. They don't have mappings.
-        realm::util::encryption_read_barrier(m_mappings[0], 0, sizeof(Header));
+#if REALM_ENABLE_ENCRYPTION
+    if (auto ref_translation_ptr = m_ref_translation_ptr.load(std::memory_order_acquire)) {
+        char* addr = ref_translation_ptr[0].mapping_addr;
+        realm::util::encryption_read_barrier(addr, NodeHeader::header_size, ref_translation_ptr[0].encrypted_mapping,
```

Comment:
I don't understand the use of `NodeHeader::header_size` and `NodeHeader::get_byte_size_from_header` here. `addr` is pointing to a `Header`, which is unrelated to `NodeHeader`, and `get_byte_size_from_header()` is reading the two top refs and interpreting it as a node header. The end result is that the barrier is only for 8 bytes and then the code below reads data past that point, which happens to work due to the mapping being page-based.

Reply:
Yes, this is a copy/paste error. Thank you for fixing this.